grpc/execution/test_utils.go

```diff
@@ -0,0 +1,197 @@
+		g.SetExtra([]byte("test"))
+		tx, _ := types.SignTx(types.NewTransaction(testNonce, common.HexToAddress("0x9a9070028361F7AAbeB3f2F2Dc07F82C4a98A02a"), big.NewInt(1), params.TxGas, big.NewInt(params.InitialBaseFee*2), nil), types.LatestSigner(&config), testKey)
+		g.AddTx(tx)
+		testNonce++
+	}
+	_, blocks, _ := core.GenerateChainWithGenesis(genesis, engine, n, generate)
+
+	if !merged {
+		totalDifficulty := big.NewInt(0)
+		for _, b := range blocks {
+			totalDifficulty.Add(totalDifficulty, b.Difficulty())
+		}
+		config.TerminalTotalDifficulty = totalDifficulty
+	}
+
+	return genesis, blocks, bridgeAddressKey, feeCollectorKey
+}
+
+func getFreePort() (int, error) {
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+	if err != nil {
+		return 0, err
+	}
+
+	l, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		return 0, err
+	}
+	defer l.Close()
+	return l.Addr().(*net.TCPAddr).Port, nil
+}
+
+// startEthService creates a full node instance for testing.
+func startEthService(t *testing.T, genesis *core.Genesis) (*node.Node, *eth.Ethereum) {
+	freePort, err := getFreePort()
+	if err != nil {
+		t.Fatal("can't get free port:", err)
+	}
+	n, err := node.New(&node.Config{
+		P2P: p2p.Config{
+			ListenAddr:  "0.0.0.0:0",
+			NoDiscovery: true,
+			MaxPeers:    25,
+		},
+		GRPCHost: "127.0.0.1",
+		GRPCPort: freePort,
+	})
+	if err != nil {
+		t.Fatal("can't create node:", err)
+	}
+
+	ethcfg := &ethconfig.Config{Genesis: genesis, SyncMode: downloader.FullSync, TrieTimeout: time.Minute, TrieDirtyCache: 256, TrieCleanCache: 256}
+	ethservice, err := eth.New(n, ethcfg)
+	if err != nil {
+		t.Fatal("can't create eth service:", err)
+	}
+
+	ethservice.SetEtherbase(testAddr)
+	ethservice.SetSynced()
+
+	return n, ethservice
+}
+
+func setupExecutionService(t *testing.T, noOfBlocksToGenerate int) (*node.Node, *eth.Ethereum, *ExecutionServiceServerV1Alpha2) {
+	t.Helper()
+	genesis, blocks, bridgeAddressKey, feeCollectorKey := generateMergeChain(noOfBlocksToGenerate, true)
+	n, ethservice := startEthService(t, genesis)
+
+	serviceV1Alpha1, err := NewExecutionServiceServerV1Alpha2(ethservice)
+	if err != nil {
+		t.Fatal("can't create execution service:", err)
+	}
+
+	feeCollector := crypto.PubkeyToAddress(feeCollectorKey.PublicKey)
+	if serviceV1Alpha1.nextFeeRecipient != feeCollector {
+		t.Fatalf("nextFeeRecipient not set correctly")
+	}
+
+	bridgeAsset := sha256.Sum256([]byte(genesis.Config.AstriaBridgeAddressConfigs[0].AssetDenom))
+	_, ok := serviceV1Alpha1.bridgeAllowedAssetIDs[bridgeAsset]
+	if !ok {
+		t.Fatalf("bridgeAllowedAssetIDs does not contain bridge asset id")
+	}
+
+	bridgeAddress := crypto.PubkeyToAddress(bridgeAddressKey.PublicKey)
+	_, ok = serviceV1Alpha1.bridgeAddresses[string(bridgeAddress.Bytes())]
+	if !ok {
+		t.Fatalf("bridgeAddress not set correctly")
+	}
+
+	utils.RegisterGRPCExecutionService(n, serviceV1Alpha1, n.Config())
+
+	if err := n.Start(); err != nil {
+		t.Fatal("can't start node:", err)
+	}
```

Comment:
why do we have to actually start the node here?

Reply:
Since all the server tests require it to start, i have just abstracted it into one place. The validation_test doesn't need it and I can add a flag to only start it when needed.